livekit-agents/livekit/agents/tts/tts.py

```diff
@@ -82,7 +89,13 @@ async def aclose(self) -> None:
         self._event_ch.close()
 
     async def __anext__(self) -> SynthesizedAudio:
-        return await self._event_ch.__anext__()
+        if self._timeout > 0:
+            try:
+                return await asyncio.wait_for(self._event_ch.__anext__(), self._timeout)
+            except asyncio.TimeoutError as e:
+                raise e.__class__("synthesis timed out")
```

Comment:
reraise instead?

Can we also make sure we have the same ErrorType when the plugins client timeout, otherwise the users may end up with 2 different errors to catch for the same timeout errors.